src/construction.cpp

```diff
@@ -1247,6 +1247,18 @@ void check_constructions()
     }
 }
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
```

Comment:
Can't you just change the signature to
```
void construction_uistatedata::serialize(JsonOut &) const
```
?

Reply:
I agree, after looking in the repository more carefully as well as doing some testing and running make astyle, it seems that preferred way of this repository is
```
( void ) json; // unused
```

I have refactored my old branch and made a new one (https://github.com/PatrikStridvall/Cataclysm-DDA/tree/crafting-uistate-plus-refactoring2). Not sure  how to update this Pull Request. Will now try to find out.

Reply:
> I hope that this doesn't confuse the automatic build process.. or the merge for that matter.

It's fine most of the time. When it's not, you could try the magical phrase "Jenkins, rebuild" which forces the bot to requeue the PR. Or just make a dummy commit.